src/errors/globalErrorHandler.ts

```diff
@@ -0,0 +1,49 @@
+import { types } from 'node:util'
+
+import { pino, stdSerializers } from 'pino'
+
+import { hasMessage } from '../utils/typeUtils'
+
+export const globalLogger = pino()
+
+export function resolveGlobalErrorLogObject(err: unknown, correlationID?: string) {
+  if (types.isNativeError(err)) {
+    return {
+      ...stdSerializers.err(err),
+      correlationID: correlationID,
+    }
+  }
+
+  if (hasMessage(err)) {
+    return correlationID ? `${err.message} (${correlationID})` : err.message
+  }
+
+  return 'Unknown global error'
+}
+
+export function executeAndHandleGlobalErrors<T>(operation: () => T) {
+  try {
+    const result = operation()
+    return result
+  } catch (err) {
+    const logObject = resolveGlobalErrorLogObject(err)
+    globalLogger.error(logObject)
+    process.exit(1)
+  }
+}
+
+export async function executeAsyncAndHandleGlobalErrors<T>(
```

Comment:
🟢 This reads a bit wrong. `<verb><adjective>` is more intuitive than `<adjective><verb>` for me. How about `executeAndHandleGlobalErrorsAsync`?

Reply:
Meditated upon this topic, and from my perspective, only operation is executed async, everything else is sync, hence original name is correct